bids/layout/layout.py

```diff
@@ -354,22 +352,56 @@ def do_begin(conn):
 
         self.session = sa.orm.sessionmaker(bind=engine)()
 
-    def _init_db(self, database_file=None, reset_database=False):
+    def _make_db_paths(self, database_path):
+        if database_path is not None:
+            database_file = os.path.join(database_path, 'layout_index.sqlilte')
+            database_sidecar = os.path.join(database_path, 'layout_args.json')
+            os.makedirs(database_path, exist_ok=True)
+        else:
+            database_file = None
+            database_sidecar = None
+        return database_file, database_sidecar
+
+    def _sanitize_instance_args(self):
+        instance_args = {
+            a: self.__dict__[a] for a in
+            ['root', 'validate', 'absolute_paths', 'regex_search', 'config_file']
+            if a in self.__dict__
+        }
+        for k in ['ignore', 'force_index']:
+            instance_args[k] = [str(a) for a in self.__dict__[k]]
+        return instance_args
+
+    def _init_db(self, database_path=None, reset_database=False):
+        database_file, database_sidecar = self._make_db_paths(database_path)
         # Reset database if needed and return whether or not it was reset
         # determining if the database needs resetting must be done prior
-        # prior to setting the session (which creates the empty database file)
+        # to setting the session (which creates the empty database file)
         reset_database = (
             reset_database or  # Manual Request
-            not database_file or  # In memory transient database
+            not database_path or  # In memory transient db
             not os.path.exists(database_file)  # New file based db created
         )
 
         self._set_session(database_file)
 
-        if reset_database:
+        instance_args = self._sanitize_instance_args()
+
+        if not reset_database:
+            saved_args = json.load(open(database_sidecar))
```

Comment:
We'll get an open filehandle warning.

```suggestion
            with open(database_sidecar) as fobj:
                saved_args = json.load(fobj)
```